dataset/util.cpp

```diff
@@ -14,8 +16,10 @@ template <class T>
 scipp::index size_of_bucket_impl(const VariableConstView &view) {
   const auto &[indices, dim, buffer] = view.constituents<T>();
   const auto &[begin, end] = unzip(indices);
+  // Using max to avoid division by zero. Other factors should be zero in this
+  // case, so this should not influence the result.
   const auto scale = sum(end - begin).template value<scipp::index>() /
```

Comment:
A minor thing:
> Other factors should be zero in this case

How about using these factors instead:
```.cpp
const auto distance = sum(end - begin).template value<scipp::index>();
const auto scale = distance != 0 ? distance / static_cast<double>(buffer.dims()) : 0;
```
If the assumption turns out to be false, this would still trigger the exception in Python.